### PR TITLE
Remove propTypes definitions from Reviews By Category

### LINKED ISSUE #9520: Convert `edit.js` to `edit.tsx` and replace `propTypes` with TypeScript definitions
## Problem

[`assets/js/blocks/reviews/reviews-by-category/edit.js`](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/blocks/reviews/reviews-by-category/edit.js) is a JS file and uses `propTypes` definitions.

## Expected outcome

1. Convert the file from `edit.js` to `edit.tsx`.
2. Replace `propTypes` with TypeScript definitions.

### PR DESCRIPTION
**Pull request by hritikchaudhary**
Fixes #9520 

### Changes in the PR:
Removed propTypes definitions from Reviews By Category.

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [x] E2E tests

#### User Facing Testing

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

**Comment by hritikchaudhary** [comment]
@nielslange I have made some file changes for this one and moved the entire code from edit to block tsx file which earlier didn't exist to keep consistency with the other blocks. I hope that is not a problem..

**Comment by nielslange** [comment]
> @nielslange I have made some file changes for this one and moved the entire code from edit to block tsx file which earlier didn't exist to keep consistency with the other blocks. I hope that is not a problem..

That's great, @hritikchaudhary. Keeping the code consistent increases readability and maintainability.

File: assets/js/blocks/reviews/reviews-by-category/block.tsx [hunk 1]
```diff
@@ -10,20 +10,20 @@ import {
...
  */
 import EditorContainerBlock from '../editor-container-block.js';
-import NoReviewsPlaceholder from './no-reviews-placeholder.js';
+import NoReviewsPlaceholder from './no-reviews-placeholder';
 import {
 	getBlockControls,
 	getSharedReviewContentControls,
 	getSharedReviewListControls,
 } from '../edit-utils.js';
+import { ReviewsByCategoryEditorProps } from './types';
```
Comment (nielslange) [inline assets/js/blocks/reviews/reviews-by-category/block.tsx]:
```suggestion
import type { ReviewsByCategoryEditorProps } from './types';
```

Given that `ReviewsByCategoryEditorProps` is a type, using `import type ...` can lead to a slightly faster compile time.

File: assets/js/blocks/reviews/reviews-by-category/edit.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import Block from './block';
+import { ReviewsByCategoryEditorProps } from './types';
```
Comment (nielslange) [inline assets/js/blocks/reviews/reviews-by-category/edit.tsx]:
```suggestion
import type { ReviewsByCategoryEditorProps } from './types';
```

Given that `ReviewsByCategoryEditorProps` is a type, using `import type ...` can lead to a slightly faster compile time.

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @hritikchaudhary. The changes look great and all test on #9629 passed. I just left two minor suggestions. Please let me know your thoughts on them.

**Comment by hritikchaudhary** [comment]
> Thanks for working on this, @hritikchaudhary. The changes look great and all test on #9629 passed. I just left two minor suggestions. Please let me know your thoughts on them.

Hi @nielslange Yes using explicitly mentioning type would lead to slightly faster compile time.. as I understand this should allow the compiler to skip the processing of the runtime code, focusing solely on the type information. 
I'll push this change np..

**Review by nielslange — APPROVED** [review]
Thanks a lot for this contribution, @hritikchaudhary. Similar to your other contributions, the changes in this PR look great and all checks on #9693 passed. 🙌

**Pull request merged**